tests/test-plugin/test_plugin/dataframe.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, TypeAlias
+
+import daft
+import daft.exceptions
+import daft.functions
+
+from narwhals._utils import (
+    Implementation,
+    ValidateBackendVersion,
+    Version,
+    not_implemented,
+)
+from narwhals.typing import CompliantLazyFrame
+
+DictFrame: TypeAlias = dict[str, list[Any]]
+
+if TYPE_CHECKING:
+    from typing_extensions import Self, TypeIs
+
+    from narwhals.dataframe import LazyFrame
+
+
+class DictLazyFrame(
+    CompliantLazyFrame[Any, "DictFrame", "LazyFrame[DictFrame]"],  # pyright: ignore[reportInvalidTypeArguments]
```

Comment:
`DictFrame` (which is just `dict[str, list[Any]]`) raises a typing error here, because it doesn't conform to `NativeLazyFrame` (which is too closely tied to the "core" dataframes)

I'm not sure what we should do here